### ISSUE TITLE
TOTP setup dialog should strip newlines

### ISSUE THREAD

**Issue by stefansundin** [keepassxreboot/keepassxc#6599]
## Overview

Totp setup dialog should strip newlines.

## Steps to Reproduce

Some websites provide the TOTP secret with this HTML markup:

```html
<div>
  <p>ABCD EFGH IJKL MNOP</p>
</div>
```

If you triple-click the TOTP, copy it, and then paste into the TOTP setup dialog, then it will give the error "You have entered an invalid secret key. The key must be in Base32 format."

If you paste it into a text document you will see that there are two newlines before the code, and one newline after it:

```


ABCD EFGH IJKL MNOP

```

## Expected Behavior

This code should also remove newlines.

https://github.com/keepassxreboot/keepassxc/blob/11afd7311725e345482f75f279fa069c552fd7ad/src/gui/TotpSetupDialog.cpp#L47-L49

## Actual Behavior

"You have entered an invalid secret key. The key must be in Base32 format."

## Context

I have encountered this problem on several websites. For example, Twitter has this problem.

## Debug info

KeePassXC - Version 2.6.4
Revision: b93fbd1

Qt 5.15.2
Debugging mode is disabled.

Operating system: macOS 11.4
CPU architecture: arm64
Kernel: darwin 20.5.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (only unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
- libgcrypt 1.9.1

**Comment by droidmonkey** [comment]
Are you sure it is not the spaces?

**Comment by stefansundin** [comment]
> Are you sure it is not the spaces?

Yep. If I delete the newlines then it works. Spaces and `=` are already stripped (`.remove(" ").remove("=")`).

**Comment by droidmonkey** [comment]
The text edit box itself should not allow line breaks. This might also be a qt bug

**Comment by stefansundin** [comment]
Hey @droidmonkey. Just wanted to let you know about another case where this fix does not work well enough.

By the way, the TOTP secret in this comment is not valid anymore. I redacted the QR code.

I am running KeePassXC 2.6.6 on Windows, and I recently used the NVIDIA GeForce Experience program to set up TOTP (in the Windows application, not in a web browser). The program presented this window:

![nvidia geforce experience](https://user-images.githubusercontent.com/1991151/123319696-978afe00-d4e5-11eb-9d8e-91031c61eb17.png)

Selecting the text and copy pasting gives me this:

```
 LJWF  MNLY  IU3F  MNTM 
 OZRG  E53J  NRCF  G52L 
 
```

And pasting that in the TOTP setup dialog gives me the same error as before ("You have entered an invalid secret key. The key must be in Base32 format."). The problem is probably that there's a newline in the middle of the code, and not at the beginning or end. I think if the fix would used `remove("\n")` instead of `trimmed()` then I think it would work in this case too.

I simply removed the newline in the middle and it worked.
